enterprise/server/remote_execution/runner/runner.go

```diff
@@ -437,14 +437,14 @@ func (r *commandRunner) Remove(ctx context.Context) error {
 		if err := r.shutdown(ctx); err != nil {
 			errs = append(errs, err)
 		}
-		if err := r.Container.Remove(ctx); err != nil {
-			errs = append(errs, err)
-		}
 		if r.stopPersistentWorker != nil {
 			if err := r.stopPersistentWorker(); err != nil {
 				errs = append(errs, err)
 			}
 		}
+		if err := r.Container.Remove(ctx); err != nil {
+			errs = append(errs, err)
+		}
```

Comment:
This had to be moved down in order to get the persistent worker's long-running `Exec()` call to return before we can call `Remove()` on it; otherwise there's a deadlock.